### PR TITLE
Fix ServiceEntries using DNS_ROUND_ROBIN

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

Fixes https://github.com/istio/istio/issues/42184

### CODE REVIEW

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -3363,10 +3363,6 @@ var ValidateServiceEntry = registerValidateFunc("ValidateServiceEntry",
 					}
 				}
 			}
-			if serviceEntry.Resolution == networking.ServiceEntry_DNS_ROUND_ROBIN && len(serviceEntry.Endpoints) != 1 {
```
Comment (ramaraochavali) [inline pkg/config/validation/validation.go:3366]: Do not we need this validation if endpoints are specified? i.e. if they are specified - should n't it be exactly one?

Comment (hzxuzhonghu) [reply pkg/config/validation/validation.go:3366]: Why cannot it have two?

Comment (hzxuzhonghu) [reply pkg/config/validation/validation.go:3366]: Does envoy deny it?

Comment (ramaraochavali) [reply pkg/config/validation/validation.go:3366]: yes. It NACKs it with this message ""LOGICAL_DNS clusters must have a single locality_lb_endpoint and a single lb_endpoint"

Comment (jacob-delgado) [reply pkg/config/validation/validation.go:3366]: 0 is allowed as well as 1. I can change this.

Comment (jacob-delgado) [reply pkg/config/validation/validation.go:3366]: I thought it'd be best to revert and come up with a fix. But I can do both.

**Review by howardjohn — COMMENTED** [review]
confused how we fixed things when only tests and validatio changed...?

**Comment by howardjohn** [comment]
Ah we are breaking >1 endpoints to fix 0 endpoints? Can we just check there is 0 or 1 endpoints instead?

**Comment by ramaraochavali** [comment]
Can you please add a test that tests zero endpoints SE is a valid one?

**Review by ramaraochavali — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Review by linsun — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

File: releasenotes/notes/42184.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: traffic-management
+issue:
+  - 42184
+
+releaseNotes:
+- |
+  **Fixed** ServiceEntries using DNS_ROUND_ROBIN being able to specify 0 endpoints
```
Comment (ericvn) [inline releasenotes/notes/42184.yaml:9]: Nit: add a period to end of sentence.
